### ISSUE TITLE
Can not install ldoc on Manjaro

### ISSUE THREAD

**Issue by Grausicht** [luarocks/luarocks#1393]
### Please read the following before submitting:
- Please do NOT submit bug reports for questions. Ask questions in the [Gitter
  chat-room](https://gitter.im/luarocks/luarocks).
- Please do NOT submit duplicate bug reports. Look through the [issue
  tracker](https://github.com/luarocks/luarocks/issues) before submitting.

### Please fill out the following:
- **Platform:**
  - Windows/Linux/MacOS.

Manjaro Linux 21.2.3

- **LuaRocks version:**
  - `luarocks --version`

3.8.0

- **Configuration file:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find
    the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com.

- **LuaRocks output from when the issue occurred:**
  - Use the `--verbose` flag.

fs.current_dir()
fs.current_dir()
fs.is_superuser()
fs.is_superuser()
fs.check_command_permissions(table: 0x5650d3750510)
fs.exists("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_writable("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_dir("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.absolute_name("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/bin")
fs.is_writable("/home/christian/.luarocks/bin")
fs.is_dir("/home/christian/.luarocks/bin")
fs.absolute_name("/home/christian/.luarocks/bin")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.make_dir("/home/christian/.cache/luarocks/https___luarocks.org")
fs.download("https://luarocks.org/manifest-5.4.zip", "/home/christian/.cache/luarocks/https___luarocks.org/manifest-5.4.zip", true)
fs.absolute_name("/home/christian/.cache/luarocks/https___luarocks.org/manifest-5.4.zip")
fs.current_dir()
HEAD	https://luarocks.org/manifest-5.4.zip
fs.absolute_name("/home/christian/.cache/luarocks/https___luarocks.org/manifest-5.4.zip")
fs.current_dir()
Installing https://luarocks.org/ldoc-1.4.6-2.rockspec
fs.check_command_permissions(table: 0x5650d3750510)
fs.exists("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_writable("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_dir("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.absolute_name("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/bin")
fs.is_writable("/home/christian/.luarocks/bin")
fs.is_dir("/home/christian/.luarocks/bin")
fs.absolute_name("/home/christian/.luarocks/bin")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.check_command_permissions(table: 0x5650d3750510)
fs.exists("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_writable("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.is_dir("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.absolute_name("/home/christian/.luarocks/lib/luarocks/rocks-5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.exists("/home/christian/.luarocks/bin")
fs.is_writable("/home/christian/.luarocks/bin")
fs.is_dir("/home/christian/.luarocks/bin")
fs.absolute_name("/home/christian/.luarocks/bin")
fs.current_dir()
fs.exists("/home/christian/.luarocks/share/lua/5.4")
fs.is_writable("/home/christian/.luarocks/share/lua/5.4")
fs.is_dir("/home/christian/.luarocks/share/lua/5.4")
fs.absolute_name("/home/christian/.luarocks/share/lua/5.4")
fs.current_dir()
fs.make_temp_dir("luarocks-rockspec-ldoc-1.4.6-2")
fs.system_temp_dir()
fs.change_dir("/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240")
fs.make_dir("/home/christian/.cache/luarocks/https___luarocks.org")
fs.download("https://luarocks.org/ldoc-1.4.6-2.rockspec", "/home/christian/.cache/luarocks/https___luarocks.org/ldoc-1.4.6-2.rockspec", true)
fs.absolute_name("/home/christian/.cache/luarocks/https___luarocks.org/ldoc-1.4.6-2.rockspec")
fs.current_dir()
HEAD	https://luarocks.org/ldoc-1.4.6-2.rockspec
fs.copy("/home/christian/.cache/luarocks/https___luarocks.org/ldoc-1.4.6-2.rockspec", "/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec")
fs.set_permissions("/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec", "read", "all")
fs._unix_moderate_permissions("644")
fs._unix_umask()

io.popen: 	umask -S
fs.execute("chmod", "644", "/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec")
fs.Q("644")
fs.Q("/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec")
fs.execute_string("chmod '644' '/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec'")

os.execute: 	chmod '644' '/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec'
Results: 3
  1 (boolean): true
  2 (string): exit
  3 (number): 0
fs.pop_dir()
fs.absolute_name("/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240/ldoc-1.4.6-2.rockspec")
fs.current_dir()
fs.make_temp_dir("luarocks-source-ldoc-1.4.6-2")
fs.system_temp_dir()
fs.change_dir("/tmp/luarocks_luarocks-source-ldoc-1.4.6-2-9720846")
fs.download("http://stevedonovan.github.io/files/ldoc-1.4.6.zip", "ldoc-1.4.6.zip", nil)
fs.absolute_name("ldoc-1.4.6.zip")
fs.current_dir()
GET	http://stevedonovan.github.io/files/ldoc-1.4.6.zip

Error: LuaRocks 3.8.0 bug (please report at https://github.com/luarocks/luarocks/issues).
Arch.: linux-x86_64
/usr/share/lua/5.4/socket/http.lua:54: bad argument #1 to 'receive' (string expected, got light userdata)
stack traceback:
	[C]: in function 'socket.http.request'
	/usr/share/lua/5.4/luarocks/fs/lua.lua:729: in upvalue 'request'
	/usr/share/lua/5.4/luarocks/fs/lua.lua:837: in upvalue 'http_request'
	/usr/share/lua/5.4/luarocks/fs/lua.lua:897: in function 'luarocks.fs.lua.download'
	(...tail calls...)
	/usr/share/lua/5.4/luarocks/fetch.lua:152: in function 'luarocks.fetch.fetch_url'
	/usr/share/lua/5.4/luarocks/fetch.lua:209: in function 'luarocks.fetch.fetch_url_at_temp_dir'
	/usr/share/lua/5.4/luarocks/fetch.lua:447: in function 'luarocks.fetch.get_sources'
	(...tail calls...)
	/usr/share/lua/5.4/luarocks/build.lua:153: in upvalue 'fetch_and_change_to_source_dir'
	/usr/share/lua/5.4/luarocks/build.lua:372: in function 'luarocks.build.build_rockspec'
	(...tail calls...)
	/usr/share/lua/5.4/luarocks/cmd/build.lua:154: in function 'luarocks.cmd.build.command'
	(...tail calls...)
	[C]: in function 'xpcall'
	/usr/share/lua/5.4/luarocks/cmd.lua:666: in function 'luarocks.cmd.run_command'
	/usr/bin/luarocks:38: in main chunk
	[C]: in ?
fs.change_dir_to_root()
fs.delete("/tmp/luarocks_luarocks-source-ldoc-1.4.6-2-9720846")
fs.delete("/tmp/luarocks_luarocks-rockspec-ldoc-1.4.6-2-9897240")

- **Description:**
  - The steps you took in plain English to reproduce the problem.

luarocks install ldoc

OR

luarocks install ldoc --local

**Comment by daurnimator** [comment]
This is likely https://bugs.archlinux.org/task/73488?project=5&string=lua-socket / https://github.com/diegonehab/luasocket/issues/331
